### PR TITLE
Add support for Tom's lossless Audio Kompressor (TAK)

### PR DESCRIPTION
**Pull request by phw**
This ads support for Tom's lossless Audio Kompressor (TAK) files with APEv2 tags.

A basic version of this has been laying around in Picard for ages (see https://github.com/metabrainz/picard/blob/master/picard/formats/mutagenext/tak.py). I extended this with reading technical details from the stream (channels, samplerate, bits per sample, encoder version and length).

### CODE REVIEW

File: mutagen/tak.py [hunk 1]
```diff
@@ -0,0 +1,237 @@
+# -*- coding: utf-8 -*-
+# Tom's lossless Audio Kompressor (TAK) reader/tagger
+#
+# Copyright 2019 Philipp Wolfer <ph.wolfer@gmail.com>
+# Copyright 2008 Lukáš Lalinský <lalinsky@gmail.com>
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 2 as
+# published by the Free Software Foundation.
```
Comment (lazka) [inline mutagen/tak.py]: To avoid any uncertainty, we would need "GPLv2 or later", not GPLv2.

File: mutagen/tak.py [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+    Attributes:
+        info (`TAKInfo`)
+    """
+
+    _Info = TAKInfo
+    _mimes = ["audio/x-tak"]
+
+    @staticmethod
+    def score(filename, fileobj, header):
+        return header.startswith(b"tBaK") + filename.lower().endswith(".tak")
```
Comment (lazka) [inline mutagen/tak.py]: use _compat.endswith(), filenames can be bytes

File: mutagen/tak.py [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+class TAKInfo(StreamInfo):
+
+    """TAK stream information.
+
+    Attributes:
+      channels (`int`): number of audio channels
+      length (`float`): file length in seconds, as a float
+      sample_rate (`int`): audio sampling rate in Hz
+      bits_per_sample (`int`): audio sample size
+      encoder_version (tuple[int]): TAK encoder version as a tuple
```
Comment (lazka) [inline mutagen/tak.py]: Somwhat similar to OptimFROG, can we make this an `encoder_info` string?

**Comment by phw** [comment]
I updated the version to GPL 2.0 or later.

@lalinsky Just calling you in because you were the author of the original code committed in 3cc216a6ef0e226c2dbc2f2a12dbe520ee1f0b2d. Given that Picard is licensed GPL 2.0 or later and you already have agreed to the version change in mutagen it is probably safe to assume that this change is in your sense, but if you object please let us know ;)

**Pull request merged**